### PR TITLE
[shootingproblem][updatenode] fix doc and node number

### PR DESCRIPTION
**Pull request by proyan**
Fix bug in shootingproblem:`updateModel`, `updateNode`

### CODE REVIEW

**Review by cmastalli — COMMENTED** [review]
I think you made a mistake.
`T_` is the number of running nodes, not the total number of nodes.

So,
  - `i == T_` is the terminal node
  - `i > T_ + 1` (or `i >= T_`) is illegal

Furthermore, I am intensively using this function in my MPC algorithm.
But the documentation is wrong:
it should be
```c++
param[in] i      node index \f$(0\leq i \leq T)\f$
```

**Comment by proyan** [comment]
Hi Carlos, I'll go point by point.
> `T_` is the number of running nodes, not the total number of nodes.

Yes
>     * `i == T_` is the terminal node

Yes

>     * `i > T_ + 1` (or `i >= T_`) is illegal

1)  `i > T_ + 1` is not the same as  `i >= T_`
2)  `i > T_ + 1` is illegal
3)  `i >= T_ + 1` is illegal
4)  `i > T_` is illegal
5)  `i >= T_ + 1` is the same as `i > T_`
6) `i >= T_` is not illegal, since `i==T_` is allowed

> But the documentation is wrong. it should be
> 
> ```c++
> param[in] i      node index \f$(0\leq i \leq T)\f$
> ```
1) Yes. the documentation was wrong.
2) `0<=i<=T_` is the same as `0<=i<T+1`

File: include/crocoddyl/core/optctrl/shooting.hxx [hunk 1]
```diff
@@ -317,7 +317,7 @@ void ShootingProblemTpl<Scalar>::circularAppend(boost::shared_ptr<ActionModelAbs
 template <typename Scalar>
 void ShootingProblemTpl<Scalar>::updateNode(std::size_t i, boost::shared_ptr<ActionModelAbstract> model,
                                             boost::shared_ptr<ActionDataAbstract> data) {
-  if (i > T_ + 1) {
+  if ((i >= T_ + 1) || (i < 0)) {
```
Comment (cmastalli) [inline include/crocoddyl/core/optctrl/shooting.hxx:320]: There is a minimal detail here. `std::size_t` is an unsigned integer, therefore, we don't need to include `|| (i < 0)`.

**Comment by proyan** [comment]
@nim65s The gitlab-CI fails with the following error:

```
WARNING: Uploading artifacts as "archive" to coordinator... failed  id=95511 responseStatus=502 Bad Gateway status=502 token=g6pTXMFG
WARNING: Retrying...                                context=artifacts-uploader error=invalid argument
WARNING: Uploading artifacts as "archive" to coordinator... failed  id=95511 responseStatus=502 Bad Gateway status=502 token=g6pTXMFG
WARNING: Retrying...                                context=artifacts-uploader error=invalid argument
WARNING: Uploading artifacts as "archive" to coordinator... failed  id=95511 responseStatus=502 Bad Gateway status=502 token=g6pTXMFG
FATAL: invalid argument                            
Cleaning up file based variables
00:00
ERROR: Job failed: exit code 1
```
Do you know why this is?

File: include/crocoddyl/core/optctrl/shooting.hxx (hunk 1 above)
Comment (proyan) [reply include/crocoddyl/core/optctrl/shooting.hxx]: DOne.

**Review by cmastalli — APPROVED** [review]

**Comment by cmastalli** [comment]
> Do you know why this is?

No, I don't but @nim65s might have an idea.

**Pull request merged**